cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/pipeline/ApplicationVerificationStage.java

```diff
@@ -211,27 +211,31 @@ private void verifyWorkflowNode(ApplicationSpecification appSpec, WorkflowSpecif
   }
 
   private void verifyWorkflowFork(ApplicationSpecification appSpec, WorkflowSpecification workflowSpec,
-                                  WorkflowNode node) {
+                                  WorkflowNode node, Set<String> existingNodeNames) {
     WorkflowForkNode forkNode = (WorkflowForkNode) node;
     Preconditions.checkNotNull(forkNode.getBranches(), String.format("Fork is added in the Workflow '%s' without" +
                                                                        " any branches", workflowSpec.getName()));
 
     for (List<WorkflowNode> branch : forkNode.getBranches()) {
-      verifyWorkflowBranch(appSpec, workflowSpec, branch);
+      verifyWorkflowNodeList(appSpec, workflowSpec, branch, existingNodeNames);
     }
   }
 
   private void verifyWorkflowCondition(ApplicationSpecification appSpec, WorkflowSpecification workflowSpec,
-                                       WorkflowNode node) {
+                                       WorkflowNode node, Set<String> existingNodeNames) {
     WorkflowConditionNode condition = (WorkflowConditionNode) node;
-    verifyWorkflowBranch(appSpec, workflowSpec, condition.getIfBranch());
-    verifyWorkflowBranch(appSpec, workflowSpec, condition.getElseBranch());
+    verifyWorkflowNodeList(appSpec, workflowSpec, condition.getIfBranch(), existingNodeNames);
+    verifyWorkflowNodeList(appSpec, workflowSpec, condition.getElseBranch(), existingNodeNames);
   }
 
-  private void verifyWorkflowBranch(ApplicationSpecification appSpec, WorkflowSpecification workflowSpec,
-                                    List<WorkflowNode> branch) {
-    for (WorkflowNode n : branch) {
-      verifyWorkflowNode(appSpec, workflowSpec, n);
+  private void verifyWorkflowNodeList(ApplicationSpecification appSpec, WorkflowSpecification workflowSpec,
+                                    List<WorkflowNode> nodeList, Set<String> existingNodeNames) {
```

Comment:
mis-alignment. 

Also, we don't really need a ``Set`` here, since you're doing the contains check below, right? If so, maybe use a ``List``, so its clear that its not require a ``Set``